homeassistant/components/godice/config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Config flow for godice."""
+
+from __future__ import annotations
+
+from homeassistant.components.bluetooth import BluetoothServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ADDRESS, CONF_NAME
+
+from .const import DOMAIN
+
+
+class GoDiceConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Setup GoDice discovered via Bluetooth."""
+
+    def __init__(self) -> None:
+        """Declare fields used later on device discovery."""
+        self._discovery_info: BluetoothServiceInfo | None = None
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the user step to pick discovered device."""
+        return self.async_abort(reason="auto_discovery_only")
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> ConfigFlowResult:
+        """GoDice discovery handler."""
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        self._discovery_info = discovery_info
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(self, user_input=None) -> ConfigFlowResult:
```

Comment:
Is there a way to set the shell (D6, D12, D20, etc) being used?
A drop-down with these options would be good: https://github.com/ParticulaCode/GoDicePythonAPI/blob/main/godice/factory.py#L11-L21

Reply:
thanks for the hint! Added shell options on integration setup, please check

Reply:
Looks good. I'm not a maintainer, so I can't approve, but I'm looking forward to this getting approved and added!